test/framework/k8s_cluster.go

```diff
@@ -200,13 +200,21 @@ func (c *K8sCluster) WaitNamespaceDelete(namespace string) {
 		c.defaultRetries,
 		c.defaultTimeout,
 		func() (string, error) {
-			_, err := k8s.GetNamespaceE(c.t,
+			nsObject, err := k8s.GetNamespaceE(c.t,
 				c.GetKubectlOptions(),
 				namespace)
 			if err != nil {
-				return "Namespace " + namespace + " deleted", nil
+				if k8s_errors.IsNotFound(err) {
+					return "Namespace " + namespace + " deleted", nil
+				}
+				return "Failed to get Namespace " + namespace, err
+			}
+
+			nsLastCondition := "unknown"
+			if len(nsObject.Status.Conditions) != 0 {
+				nsLastCondition = nsObject.Status.Conditions[len(nsObject.Status.Conditions)-1].String()
 			}
-			return "Namespace available " + namespace, fmt.Errorf("Namespace %s still active", namespace)
+			return "Namespace available " + namespace, fmt.Errorf("Namespace %s still active, last condition: %s", namespace, nsLastCondition)
```

Comment:
I think we're not printing `out` from the `DoWithRetry` so we probably won't see this line no?

Reply:
We can see that, here's what I extract from the CI:

```text
Zone and Global with Helm chart should deploy Zone and Global on 2 clusters 2024-09-27T14:14:16Z retry.go:103: Wait for kuma-test Namespace to terminate. returned an error: Namespace kuma-test still active. Sleeping for 6s and will try again.
```

Reply:
but this is printing `err` not `out`?